contracts/governance/src/ZRXWrappedToken.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+
+  Copyright 2023 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/token/ERC20/ERC20.sol";
+import "@openzeppelin/token/ERC20/extensions/draft-ERC20Permit.sol";
+import "@openzeppelin/token/ERC20/extensions/ERC20Wrapper.sol";
+import "@openzeppelin/governance/utils/IVotes.sol";
+import "./IZeroExVotes.sol";
+
+contract ZRXWrappedToken is ERC20, ERC20Permit, ERC20Wrapper {
```

Comment:
Because ZRX is a well-behaved ERC20, we can override `totalSupply` in the wrapped token to call `ZRX.balanceOf(address(this))` instead of storing it in its own slot. This saves gas when minting/burning in the wrapped token.

Reply:
Good idea, just that OZ contract still writes to the `private _totalSupply` in storage and I can't overwrite that without overriding their `_mint` and `_burn` logic.